### PR TITLE
Support jupyterlab 4 and drop jupyterlab <4 and notebook <7 support

### LINKED ISSUE #9: Installation on JL 2.x returns incompatibility issues
I was trying to install this plugin in both JupyterLab 2.2.8 and 2.1.5 venvs by first 
`pip install jupyter-launcher-shortcuts` and then `sudo jupyter labextension install jupyterlab-launcher-shortcuts`however in both cases received incompatibility issues respectively:
```Conflicting Dependencies:
JupyterLab              Extension      Package
>=2.2.5 <2.3.0          >=0.19.1 <0.20.0@jupyterlab/application
>=2.2.5 <2.3.0          >=0.19.1 <0.20.0@jupyterlab/launcher
```
```
Conflicting Dependencies:
JupyterLab              Extension      Package
>=2.1.2 <2.2.0          >=0.19.1 <0.20.0@jupyterlab/application
>=2.1.1 <2.2.0          >=0.19.1 <0.20.0@jupyterlab/launcher
```
Is it a known issue or am I doing something wrong?

### LINKED ISSUE #12: Compatibility error with JupyterLab 3.0
Hi there! this extensions was working great until the realease of JupyterLab > 3.0.0
I'm getting the following error when installing it:
`Conflicting Dependencies:
JupyterLab              Extension      Package
>=3.0.3 <3.1.0          >=2.0.0 <3.0.0 @jupyterlab/application
>=3.0.2 <3.1.0          >=2.0.0 <3.0.0 @jupyterlab/launcher
See the log file for details:  /tmp/jupyterlab-debug-lc_72nit.log
The command '/bin/bash -o pipefail -c jupyter labextension install @jupyterlab/server-proxy jupyterlab-launcher-shortcuts' returned a non-zero code: 1`

Could this extension be adjusted to support the latest JupyterLab? thanks!!

### LINKED ISSUE #19: Embeding JS inside Python whell (Jupyterlab 3+ feature)
Since Jupyterlab 3+ allow to avoid npm extension by building js directly inside python wheel. 

I currently working (for fun and experiment) on a PR that aims to embed JS inside Python wheel. Interested?

Best regards,
Tristan

### PR DESCRIPTION
**Pull request by tristanlt**
This pull request is for a new version jupyter-launcher-shortcuts for jupyterlab 4+.

* Jupyterlab >=4,<5 compatibility  #12 
* migrated from notebooks module to jupyter_server python modules
* prebuild labextension to avoid build extension at install time (and make npmjs package useless)  #19
* build & package management now based on hatch (releases should be easiest #20 )
* flattened project directory

## Update by @consideratio

- Requires JupyterLab 4+, up from JupyterLab 2 I think
- Requires Python 3.8+, up from Python 3.5+
- Breaks function for classic notebook UI server (notebook v6 and v7)
  - Removes python dependency on `notebook`
  - Closes #22
  - Related to #25
- Removes unused python depdency on `simpervisor` and `aiohttp`
- Fixes #12 
- Fixes #19
- Closes #9 (because we likeley no longer support JupyterLab 2 via [aebb35f](https://github.com/2i2c-org/jupyter-launcher-shortcuts/pull/24/commits/aebb35fd73f5647735375fd7878495bb107d7a7e))

---

Changes by me:

- Updated README to mention Jupyter Notebook UI currently isn't working
- Added comment in setup.py
- Removed a copy-paste remnant file config.py
- Declared dependencies on jupyter_server, tornado, and traitlets explicitly
- Added an install.json file
  It helps couple the labextension to the python package when getting information via `jupyter labextesion list`, where the parenthesis after `enabled OK` is now being shown.
  ```
  jupyter labextension list
  JupyterLab v4.2.3
  /home/erik/miniforge3/share/jupyter/labextensions
          jupyter_launcher_shortcuts v4.1.0 enabled OK (python, jupyter-launcher-shortcuts)
  ```

### CODE REVIEW

**Comment by tristanlt** [comment]
Hi @yuvipanda ! How are you ? 

I've worked on this extension because it useful in our virtual research environment. I have to migrate our VRE to jupyterlab4 and this extension is the last lock. 

I hope you'll find some time to review the pull request. And, if you want, maybe give me the rights on pypi/conda-forge to releases packages.

Best regards,
@tristanlt

**Comment by yuvipanda** [comment]
I had somehow completely missed this, @tristanlt :( Apologies for the super super delayed response! Someone will be along shortly to merge this and make a release.

In the medium run, I'd like to move this into the jupyterlab-contrib (https://github.com/jupyterlab-contrib/) org. Thoughts on that, @jtpio?

**Comment by jtpio** [comment]
> I'd like to move this into the jupyterlab-contrib (https://github.com/jupyterlab-contrib/) org.

Sure. Would you like to click on "Transfer your extension" on this page? https://jupyterlab-contrib.github.io/

Of note, there is a new extension providing an alternative launcher, which might also be of interest: https://github.com/nebari-dev/jupyterlab-launchpad. Note sure it provides a way for users to declare additional entries (for example via the settings system), but maybe it could be requested as a feature.

File: jupyter_launcher_shortcuts/__init__.py [hunk 1]
```diff
@@ -70,3 +70,9 @@ def _jupyter_nbextension_paths():
         'src': 'static',
         "require": "jupyter_launcher_shortcuts/tree"
     }]
+
+def _jupyter_labextension_paths():
```
Comment (agoose77) [inline jupyter_launcher_shortcuts/__init__.py:74]: We should also rename `_jupyter_server_extension_paths` to `_jupyter_server_extension_points` c.f. https://github.com/jupyter-server/jupyter_server/blob/f2d40f692c29e155349b179d57a454e7f23ea898/jupyter_server/extension/utils.py#L80-L93

Comment (consideRatio) [reply jupyter_launcher_shortcuts/__init__.py:74]: Done!

File: setup.py [hunk 2]
```diff
@@ -1,26 +1,2 @@
...
-    data_files=[
-        ('share/jupyter/nbextensions/jupyter_launcher_shortcuts', glob('jupyter_launcher_shortcuts/static/*')),
-        ('etc/jupyter/jupyter_notebook_config.d', ['jupyter_launcher_shortcuts/etc/serverextension.json']),
-        ('etc/jupyter/nbconfig/tree.d', ['jupyter_launcher_shortcuts/etc/nbextension.json'])
-    ],
-    include_package_data=True,
-    zip_safe=False
-)
+# setup.py shim for use with applications that require it.
+__import__("setuptools").setup()
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply setup.py]: Done!

**Comment by consideRatio** [comment]
Thank you soo much @tristanlt for working this!!! Thank you @agoose77 for  review help and @yuvipanda for keeping track of this!

**Comment by consideRatio** [comment]
I'll go for a merge and work to get a release out soon versioned 4.0.0.

**Pull request merged**